packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1704,43 +1705,76 @@ String nextPlatform(String currentPlatform, FeatureFlags featureFlags) {
   }
 }
 
+/// A manager for a devtools debugger instance.
+///
+/// NB: This class is overriden in google3.
 class DevtoolsLauncher {
-  io.HttpServer _devtoolsServer;
+  DevtoolsLauncher({
+    @required ProcessManager processManager,
+    @required String pubExecutable,
+    @required Logger logger,
+  }) : _processManager = processManager,
+       _pubExecutable = pubExecutable,
+       _logger = logger;
+
+  final ProcessManager _processManager;
+  final String _pubExecutable;
+  final Logger _logger;
+
+  io.Process _devtoolsProcess;
+  Uri _devtoolsUri;
+
+  static final RegExp _devtoolsPattern = RegExp(r'Serving DevTools at ((http|//)[a-zA-Z0-9:/=_\-\.\[\]]+)');
+
+  /// Launch a Dart DevTools process, optionally targeting a specific VM Service URI if
+  /// [observatoryAddress] is non-null.
   Future<void> launch(Uri observatoryAddress) async {
     try {
-      await serve();
-      await devtools_server.launchDevTools(
-        <String, dynamic>{
-          'reuseWindows': true,
-        },
-        observatoryAddress,
-        'http://${_devtoolsServer.address.host}:${_devtoolsServer.port}',
-        false,  // headless mode,
-        false,  // machine mode
-      );
+      _devtoolsProcess = await _processManager.start(<String>[
```

Comment:
there is a daemon mode for devtools inspired by the one in flutter tools. Pass it in and you won't need regex match to detect what port is used.
https://github.com/flutter/devtools/blob/master/packages/devtools_server/lib/src/server.dart#L85